packages/react/src/components/Popover/Popover.tsx

```diff
@@ -176,12 +176,30 @@ const arrowPseudoElement = {
   name: 'ArrowPseudoElement',
   fn(data: MiddlewareState) {
     const { elements, rects, placement } = data;
-    const arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
-    const arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+    let arrowX = 0;
+    let arrowY = 0;
+
+    /**
+     * if popover is wider, set arrow in the middle of the reference
+     * if the reference is wider, set the arrow in the middle of the popover
+     */
+    if (rects.reference.width > rects.floating.width) {
+      arrowX = rects.floating.width / 2 - ARROW_HEIGHT / 2;
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+    } else {
+      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
+      arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
+    }
```

Comment:
```suggestion
    if (rects.reference.width > rects.floating.width) {
      arrowX = rects.floating.width / 2 - ARROW_HEIGHT / 2;
    } else {
      arrowX = rects.reference.width / 2 + rects.reference.x - data.x;
    }
    arrowY = rects.reference.height / 2 + rects.reference.y - data.y;
```

Reply:
Since `arrowY` is always the same? :)

Reply:
Suggest also removing the `else` statement for better readability as its redundant.

Reply:
Thought I disagree with the readability, I'll move the else assignment as the default.